pkg/apis/core/v1alpha1/cluster_types.go

```diff
@@ -231,3 +236,24 @@ var _ resource.StatusSubResource = &ClusterStatus{}
 func (in ClusterStatus) CopyTo(parent resource.ObjectWithStatusSubResource) {
 	parent.(*Cluster).Status = in
 }
+
+// Connection spec for an existing cluster.
+type ClusterConnectionStatus struct {
+	// Defines connection to a Kubernetes cluster.
+	Kubernetes *KubernetesClusterConnectionStatus `json:"kubernetes,omitempty" protobuf:"bytes,1,opt,name=kubernetes"`
+}
+
+// Kubernetes-specific fields for connection status
+type KubernetesClusterConnectionStatus struct {
```

Comment:
i struggled a lot with what the name of this should be...ContextStatus? something else?

i expect we'll want to do a similar thing with Docker (where we publish the resolved docker host)

Reply:
I think this makes sense!